src/components/application_manager/src/commands/mobile/on_button_event_notification.cc

```diff
@@ -106,20 +106,24 @@ void OnButtonEventNotification::Run() {
       continue;
     }
 
-    //Send ButtonEvent notification only in HMI_FULL or HMI_LIMITED mode
+    // Send ButtonEvent notification only in HMI_FULL or HMI_LIMITED mode
     if ((mobile_api::HMILevel::HMI_FULL != subscribed_app->hmi_level()) &&
         (mobile_api::HMILevel::HMI_LIMITED != subscribed_app->hmi_level())) {
       LOG4CXX_WARN_EXT(logger_, "OnButtonEvent in HMI_BACKGROUND or NONE");
       continue;
     }
 
-    //Send ButtonEvent notification for OK button only in HMI_FULL mode
-    if ((static_cast<uint32_t>(mobile_apis::ButtonName::OK) == btn_id) &&
-        (!subscribed_app->IsFullscreen())) {
-      continue;
-    }
+    if (app) {
+      // if "appID" is present  send it to Named app , only if its FULL or
+      // LIMITED
+      if (app->app_id() == subscribed_app->app_id()) {
```

Comment:
@OHerasym why are you going to check `app->app_id()` here? We're working with `subscribed_app`s